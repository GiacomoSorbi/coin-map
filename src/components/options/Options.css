.options {
  background-color: bisque;
  padding: 1vmin;
}
