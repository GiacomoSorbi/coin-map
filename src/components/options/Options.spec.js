import React from 'react';

import Options from './index';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const fn = jest.fn();
  const sampleRender1 = renderer
    .create(
      <Options
        filters={{ lat1: 0, lat2: 100, lon1: -90, lon2: 0, limit: 500 }}
        onChangeFilters={fn}
      />
    )
    .toJSON();
  expect(sampleRender1).toMatchSnapshot();
});
