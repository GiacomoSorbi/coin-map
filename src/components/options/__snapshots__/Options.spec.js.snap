// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="options"
>
  <div
    className="input-field"
  >
    <label
      htmlFor="limit"
    >
      Maximum returned matches
    </label>
    <input
      id="limit"
      max={16000}
      min={0}
      onChange={[MockFunction]}
      type="number"
      value={500}
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lat1"
    >
      Minimum Latitude
    </label>
    <input
      id="lat1"
      max={99}
      min={-90}
      onChange={[MockFunction]}
      type="number"
      value={0}
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lat2"
    >
      Maximum Latitude
    </label>
    <input
      id="lat2"
      max={90}
      min={1}
      onChange={[MockFunction]}
      type="number"
      value={100}
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lon1"
    >
      Minimum Longitude
    </label>
    <input
      id="lon1"
      max={-1}
      min={-180}
      onChange={[MockFunction]}
      type="number"
      value={-90}
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="lon2"
    >
      Maximum Longitude
    </label>
    <input
      id="lon2"
      max={180}
      min={-89}
      onChange={[MockFunction]}
      type="number"
      value={0}
    />
  </div>
</header>
`;
